pkg/controller/grafanadashboard/dashboard_controller.go

```diff
@@ -132,6 +133,53 @@ func (r *ReconcileGrafanaDashboard) Reconcile(request reconcile.Request) (reconc
 		return r.deleteDashboard(cr)
 	}
 
+	jsonTest, err := isJSON(cr.Spec.Json)
+	t := time.Now()
+
+	// JSON is invalid and we haven't set a status message yet
+	if jsonTest == false && len(cr.Status.Messages) == 0 {
+		log.Info(fmt.Sprintf("found dashboard '%s/%s' but JSON is not valid", cr.Namespace, cr.Name))
+		jsonError := err.Error()
+		msg := i8ly.GrafanaDashboardStatusMessage{
+			Message:   fmt.Sprintf("Invalid JSON, Error: %s", jsonError),
+			Timestamp: fmt.Sprintf(t.Format("2006-01-02 15:04:05")),
+		}
+		// Dashboard has been created but JSON is now invalid
+		if cr.Status.Phase == common.StatusResourceCreated {
+			cr.Status.Phase = common.StatusResourceUninitialized
+			return r.deleteDashboard(cr)
+
+		}
+		r.updateStatus(cr, msg)
+		return reconcile.Result{}, nil
+	}
+
+	// JSON is invalid but we've already set a status message
+	if jsonTest == false && len(cr.Status.Messages) > 0 {
```

Comment:
Not sure if this test will work, because there are also other status messages (unrelated to json errors) that can be appended to dashboards (e.g. when a plugin installation fails).

Reply:
What do you think about the following approach:

1. We always append the status message if we encounter the json error
1. We only process dashboards if they have changed, by setting a hash (of the last json) in their status field and we ignore dashboard reconcile events as long as the new hash is equal to the last hash?

Something like that is currently done for data sources https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/grafanadatasource/datasource_controller.go#L127 because Openshift seems to update resources every ~10 minutes for an unknown reason.

Reply:
Yeah that's a must better approach than what I fell into. I'll implement that.